ms2query/ms2library.py

```diff
@@ -0,0 +1,332 @@
+            File location of a pickled file with ms2ds embeddings in a
+            pd.Dataframe with as index the spectrum id.
+
+        **settings:
+            As additional parameters predefined settings can be changed.
+        spectrum_id_column_name:
+            The name of the column or key in dictionaries under which the
+            spectrum id is stored. Default = "spectrumid"
+        cosine_score_tolerance:
+            Setting for calculating the cosine score. If two peaks fall within
+            the cosine_score tolerance the peaks are considered a match.
+            Default = 0.1
+        base_nr_mass_similarity:
+            The base nr used for normalizing the mass similarity. Default = 0.8
+        max_parent_mass:
+            The value used to normalize the parent mass by dividing it by the
+            max_parent_mass. Default = 13428.370894192036
+        """
+        # pylint: disable=too-many-arguments
+        # Change default settings to values given in **settings
+        settings = self._set_settings(settings)
+
+        # Set given settings
+        self.spectrum_id_column_name = settings["spectrum_id_column_name"]
+        # todo create a ms2query model class that stores the model but also the
+        #  settings used, since the settings used should always be the same as
+        #  when the model was trained
+        self.cosine_score_tolerance = settings["cosine_score_tolerance"]
+        self.base_nr_mass_similarity = settings["base_nr_mass_similarity"]
+        # todo make new model that has a fixed basic mass
+        self.max_parent_mass = settings["max_parent_mass"]
+
+        # Load models and set sqlite_file_location
+        self.sqlite_file_location = sqlite_file_location
+        self.s2v_model = Word2Vec.load(s2v_model_file_name)
+        self.ms2ds_model = load_ms2ds_model(ms2ds_model_file_name)
+
+        # loads the library embeddings into memory
+        self.s2v_embeddings: pd.DataFrame = load_pickled_file(
+            pickled_s2v_embeddings_file_name)
+        self.ms2ds_embeddings: pd.DataFrame = load_pickled_file(
+            pickled_ms2ds_embeddings_file_name)
+
+    @staticmethod
+    def _set_settings(new_settings: Dict[str, Union[str, bool]],
+                      ) -> Dict[str, Union[str, float]]:
+        """Changes default settings to new_settings and creates file names
+
+        Args:
+        ------
+        new_settings:
+            Dictionary with settings that should be changed. Only the
+            keys given in default_settings can be used and the type has to be
+            the same as the type of the values in default settings.
+        """
+        # Set default settings
+        default_settings = {"spectrum_id_column_name": "spectrumid",
+                            "cosine_score_tolerance": 0.1,
+                            "base_nr_mass_similarity": 0.8,
+                            "max_parent_mass": 13418.370894192036}
```

Comment:
Where does this number come from? I guess it is the max value from the dataset we work with and is used to later scale the parent_mass to 0 - 1.0 range?
Still feels a bit odd to me because 1) it is a very high number (most compounds in the dataset are far smaller), 2) it might be a false value (the parent mass estimation can go wrong if the charge value is not correct) and 3) it will in principle be different for any other dataset. So I would rather use a generic number, say 1000.0 or 2000.0...

Reply:
Yes I agree, we already discussed this some time ago. However for the current model I used the parent mass 13418 for training, so I kept it like this for now. But I think 1000 seems like a good idea. I actually think it might be good to create a class for the MS2Query neural network, that besides the model also stores all the settings that were used when training it, since these settings should never be changed when using this model. I will make an issue for this.